chaincode/input.go

```diff
@@ -31,42 +31,46 @@ var (
 
 // inputObjective is the representation of input args to register a Objective
 type inputObjective struct {
-	Name                      string           `validate:"required,gte=1,lte=100" json:"name"`
-	DescriptionHash           string           `validate:"required,len=64,hexadecimal" json:"descriptionHash"`
-	DescriptionStorageAddress string           `validate:"required,url" json:"descriptionStorageAddress"`
-	MetricsName               string           `validate:"required,gte=1,lte=100" json:"metricsName"`
-	MetricsHash               string           `validate:"required,len=64,hexadecimal" json:"metricsHash"`
-	MetricsStorageAddress     string           `validate:"required,url" json:"metricsStorageAddress"`
-	TestDataset               inputDataset     `validate:"omitempty" json:"testDataset"`
-	Permissions               inputPermissions `validate:"required" json:"permissions"`
+	Name                      string           	`validate:"required,gte=1,lte=100" json:"name"`
+	DescriptionHash           string           	`validate:"required,len=64,hexadecimal" json:"descriptionHash"`
+	DescriptionStorageAddress string           	`validate:"required,url" json:"descriptionStorageAddress"`
+	MetricsName               string           	`validate:"required,gte=1,lte=100" json:"metricsName"`
+	MetricsHash               string           	`validate:"required,len=64,hexadecimal" json:"metricsHash"`
+	MetricsStorageAddress     string           	`validate:"required,url" json:"metricsStorageAddress"`
+	TestDataset               inputDataset     	`validate:"omitempty" json:"testDataset"`
+	Permissions               inputPermissions 	`validate:"required" json:"permissions"`
+	Metadata				  map[string]string	`validate:"omitempty" json:"metadata"`
```

Comment:
It's strange that your editor didn't take care of it but usually the different section of a struct validation are aligned.

Reply:
Got the problem: you've used tabs between names, types, and tags. It should have been spaces.

Reply:
I formatted all go file, see the diff: https://gist.github.com/thibaultrobert/1e142684add2418739b6793c0a3103b6
You can apply it or try to tweak the formatting tools of your editor/IDE.

Reply:
Yes, thank you! It should be better like this